pkg/kanctl/actionset.go

```diff
@@ -764,14 +767,38 @@ func verifyRepositoryServerParams(repoServer *crv1alpha1.ObjectReference, crCli
 			}
 			return errors.New("error while fetching repo server")
 		}
-		if rs.Status.Progress != crv1alpha1.Ready {
-			err = errors.New("Repository Server Not Ready")
-			return errors.Wrapf(err, "Please make sure that Repository Server CR '%s' is in Ready State", repoServer.Name)
+		wait, _ := cmd.Flags().GetBool(waitForReadyFlagName)
+		if wait {
+			err = waitForKopiaRepositoryServerReady(ctx, crCli, rs)
+			if err != nil {
+				return err
+			}
+		} else {
+			if rs.Status.Progress != crv1alpha1.Ready {
+				err = errors.New("Repository Server Not Ready")
+				return errors.Wrapf(err, "Please make sure that Repository Server CR '%s' is in Ready State", repoServer.Name)
+			}
 		}
```

Comment:
```suggestion
		if wait {
			return  waitForKopiaRepositoryServerReady(ctx, crCli, rs)
		}
		if rs.Status.Progress != crv1alpha1.Ready {
			err = errors.New("Repository Server Not Ready")
			return errors.Wrapf(err, "Please make sure that Repository Server CR '%s' is in Ready State", repoServer.Name)
		}
```
We can get rid of `else` block

Reply:
Done !